extensions/rv32-adapters/src/eq_mod.rs

```diff
@@ -394,9 +395,14 @@ impl<
     type ReadData = [[u8; TOTAL_READ_SIZE]; NUM_READS];
     type WriteData = [u8; RV32_REGISTER_NUM_LIMBS];
 
-    fn read<Mem>(&self, memory: &mut Mem, instruction: &Instruction<F>) -> Self::ReadData
+    fn read<Mem, Ctx>(
+        &self,
+        state: &mut VmStateMut<Mem, Ctx>,
```

Comment:
```suggestion
        state: VmStateMut<Mem, Ctx>,
```

Reply:
i'll take a look at this again after rebasing but this was causing the compiler to complain so i had to add the `&mut`

Reply:
this doesn't work because i need to pass the state as a reference now and call a callback to update `ctx.trace_heights` on every memory operation. ideally this should just be a `&mut VmState<Mem, Ctx>` but for that we'll need to take memory out of the controller

https://github.com/openvm-org/openvm/blob/7508e8278ad6fd48fe2ca45d73e5056363549068/extensions/rv32im/circuit/src/adapters/alu.rs#L304-L305
https://github.com/openvm-org/openvm/blob/7508e8278ad6fd48fe2ca45d73e5056363549068/extensions/rv32im/circuit/src/adapters/mod.rs#L93-L105